mono/utils/mono-publib.h

```diff
@@ -144,6 +144,12 @@ mono_set_allocator_vtable (MonoAllocatorVTable* vtable);
 #pragma GCC diagnostic error "-Wdeprecated-declarations"
 #endif
 
+#elif defined (_MSC_VER)
+#define MONO_RT_EXTERNAL_ONLY __declspec (deprecated ("The mono runtime must not call this function.")) \
+	MONO_RT_CENTRINEL_SUPPRESS
+// Turn deprecation warnings into errors
+#pragma warning (error : 4996)
```

Comment:
Same question as you -- what about all the other errors?
Our windows build is so full of warnings so keeping it as a warning would be worthless (which you aren't doing here, understood).

Reply:
Most of the warnings are in common code, but other platforms are building with lower warning levels or excluding several of the categories causing most of the warnings in windows builds. One option could be to reduce warning level on windows to similar levels used in gcc/clang builds and then keeping this as a warning.